### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by trowland10**
Added method for registration and fixed a couple of typos in Login

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -80,6 +80,85 @@ app.get('/resetpassword', isAuthenticated, function (req, res, next) {
...
+                data.confirmPassword = ''
+                res.render('auth/register', data)
+            } else {
+                bcrypt.hash(data.password, 10, function (err, hash) {
+                    if (err) {
+                        req.flash('error', err)
+                        res.render('auth/register', data)
+                    } else {
+                        sql.pool.query('INSERT INTO Users (firstName, lastName, email, password, tempPassword, address, city, state, postalCode, country, worldwide, points)\
+                        VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)', [data.firstName, data.lastName, data.email, hash, 0, data.address, data.city, data.state,
```
Comment (mkoeplin) [inline app.js]: Points has a default of 0 so we could probably remove it from the insert statement just so we are only inserting what is actually entered on the form. We should also probably set tempPassword to default to 0 so you can remove it from the insert as well.

File: app.js [hunk 2]
```diff
@@ -80,6 +80,85 @@ app.get('/resetpassword', isAuthenticated, function (req, res, next) {
...
+                        data.postalCode, data.country, Number(data.worldwide), 0],
```
Comment (mkoeplin) [inline app.js]: You shouldn't need to convert the data.worldwide value to a number during the insert, it doesn't hurt but I don't think it's necessary

File: views/auth/login.handlebars [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <div class="pb-3">
-  <div>New to BookSwap? <a href="/Register">Register Now</a></div>
+  <div>New to BookSwap? <a href="/register">Register Now</a></div>
```
Comment (mkoeplin) [inline views/auth/login.handlebars:2]: Routes aren't case sensitive, this is more of just an FYI so you know it doesn't have to be changed to work!

File: views/auth/register.handlebars [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            <input type="text" class="form-control" name="address" value="{{address}}" placeholder="1343 Anywhere St Apt 100" required>
+        </div>
+    </div>
+    <div class="form-row">
+        <div class="col">
+            <label for="city"> City</label>
+            <input type="text" class="form-control" name="city" value="{{city}}" required>
+        </div>
+        <div class="col">
+            <label for="state">State</label>
```
Comment (mkoeplin) [inline views/auth/register.handlebars:44]: I'd probably make state/province a free form field and create a dropdown for countries. I'd even probably create a Countries table to list the codes and display values and pull from there to populate the dropdown.

Check out this example! https://bootsnipp.com/snippets/r8W

**Review by mkoeplin — CHANGES_REQUESTED** [review]
Take a look at a few of the suggestions and let me know what you think and if you plan to make any of the changes. Otherwise everything looks good!

File: app.js (hunk 2 above)
Comment (trowland10) [reply app.js]: decided to remove this, thanks for the heads up!

File: app.js (hunk 1 above)
Comment (trowland10) [reply app.js]: Update: I changed the sql file to default temp pass to 0, removed points and temp pass from insert in the latest commit

File: views/auth/register.handlebars (hunk 4 above)
Comment (trowland10) [reply views/auth/register.handlebars:44]: Changed state to freeform and added the country dropdown. I decided to put the dropdown in HTML for now, mainly for the sake of time so you can get to test your methods

**Review by mkoeplin — APPROVED** [review]

**Pull request merged**